streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -511,15 +512,21 @@ private RepartitionTopics prepareRepartitionTopics(final Cluster metadata) {
             metadata,
             logPrefix
         );
-        final boolean isMissingInputTopics = !repartitionTopics.setup();
+        repartitionTopics.setup();
+        final boolean isMissingInputTopics = !repartitionTopics.missingSourceTopicExceptions().isEmpty();
         if (isMissingInputTopics) {
             if (!taskManager.topologyMetadata().hasNamedTopologies()) {
                 throw new MissingSourceTopicException("Missing source topics.");
+            } else {
+                for (final StreamsException missingSourceTopicException : repartitionTopics.missingSourceTopicExceptions()) {
+                    nonFatalExceptionsToHandle.add(missingSourceTopicException);
+                }
```

Comment:
nit:
IntelliJ says that this could be simplified to:

```suggestion
                nonFatalExceptionsToHandle.addAll(repartitionTopics.missingSourceTopicExceptions());
```